include/message_filters/subscriber.h

```diff
@@ -43,6 +43,7 @@
 namespace message_filters
 {
 
+template <class NodeT = rclcpp::Node>
```

Comment:
If you stored pointers to the various Node interface functions you needed, you could avoid making this class templated, and just template the methods.

See this function:

https://github.com/ros2/rclcpp/blob/98fc7fecc25b2fecc089d1117cee6803eda34836/rclcpp/include/rclcpp/create_subscription.hpp#L205-L235

You could store just the `rclcpp::node_interfaces::NodeParametersInterface::SharedPtr` and `rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr` from the `NodeT` and then call the above function to subscribe to things.